skimage/exposure/_adapthist.py

```diff
@@ -214,45 +213,64 @@ def clip_histogram(hist, clip_limit):
     excess_mask = hist > clip_limit
     excess = hist[excess_mask]
     n_excess = excess.sum() - excess.size * clip_limit
+    hist[excess_mask] = clip_limit
 
     # Second part: clip histogram and redistribute excess pixels in each bin
-    bin_incr = int(n_excess / hist.size)  # average binincrement
+    bin_incr = n_excess // hist.size  # average binincrement
     upper = clip_limit - bin_incr  # Bins larger than upper set to cliplimit
 
-    hist[excess_mask] = clip_limit
-
     low_mask = hist < upper
     n_excess -= hist[low_mask].size * bin_incr
     hist[low_mask] += bin_incr
 
-    mid_mask = (hist >= upper) & (hist < clip_limit)
+    mid_mask = np.logical_and(hist >= upper, hist < clip_limit)
     mid = hist[mid_mask]
-    n_excess -= mid.size * clip_limit - mid.sum()
+    n_excess += mid.sum() - mid.size * clip_limit
     hist[mid_mask] = clip_limit
 
-    prev_n_excess = n_excess
-
     while n_excess > 0:  # Redistribute remaining excess
-        index = 0
-        while n_excess > 0 and index < hist.size:
-            under_mask = hist < 0
-            step_size = int(hist[hist < clip_limit].size / n_excess)
-            step_size = max(step_size, 1)
-            indices = np.arange(index, hist.size, step_size)
-            under_mask[indices] = True
-            under_mask = (under_mask) & (hist < clip_limit)
-            hist[under_mask] += 1
-            n_excess -= under_mask.sum()
-            index += 1
-        # bail if we have not distributed any excess
+        prev_n_excess = n_excess
+        for index in range(hist.size):
+            under_mask = hist < clip_limit
+            step_size = max(1, np.count_nonzero(under_mask) // n_excess)
+            under_mask = under_mask[index::step_size]
+            hist[index::step_size][under_mask] += 1
+            n_excess -= np.count_nonzero(under_mask)
+            if n_excess <= 0:
+                break
         if prev_n_excess == n_excess:
             break
-        prev_n_excess = n_excess
 
     return hist
 
 
-def map_histogram(hist, min_val, max_val, n_pixels):
+def clip_histogram(hist, clip_limit):
+    """Perform clipping of the histogram and redistribution of bins.
+
+    The histogram is clipped and the number of excess pixels is counted.
+    Afterwards the excess pixels are equally redistributed across the
+    whole histogram (providing the bin count is smaller than the cliplimit).
+
+    Parameters
+    ----------
+    hist : ndarray
+        Histogram array.
+    clip_limit : int
+        Maximum allowed bin count.
+
+    Returns
+    -------
+    hist : ndarray
+        Clipped histogram.
+    """
+    warnings.warn("clip_histogram is deprecated and will be removed in "
+                  "version 0.19. Please use the rivate function "
+                  "_clip_histogram instead.", category=FutureWarning,
```

Comment:
I do not understand why this function will be deprecated and why you encourage to use a private function. Private functions must not be used by end-users and the API can change without notice.

Reply:
I simply considered these functions are not private since they do not respect the naming conventions for private functions.
But you are right, they are in a private module, so no need for explicitly renaming them.
I will revert the changes, thx.